### PR TITLE
add GitHub Actions

### PR DESCRIPTION
**Pull request by szabgab**
Surprisingly, enabling either AUTHOR_TESTING or RELEASE_TESTING will make the same tests fail. So I disabled them.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    - name: Show Errors on OSX
+      if:  ${{ failure() && startsWith( matrix.runner, 'macos-')}}
+      run: |
+         cat  /Users/runner/.cpanm/work/*/build.log
+
+    - name: Run tests
+      #env:
+      #  AUTHOR_TESTING: 1
+      #  RELEASE_TESTING: 1
```
[the remark this answers is not in the document]
Comment (szabgab) [reply .github/workflows/ci.yml:62]: s/the same/some/  in my original comment.

The failure report is here: https://github.com/szabgab/perl-opentracing-implementation-datadog/actions/runs/4600503012/jobs/8127175327

It is unclear to me how are these failures related to the environment variables.

[the remark this answers is not in the document]
Comment (vanHoesel) [reply .github/workflows/ci.yml:62]: could you run them again ?

[the remark this answers is not in the document]
Comment (szabgab) [reply .github/workflows/ci.yml:62]: Sure, I can rebase and enable them

[the remark this answers is not in the document]
Comment (szabgab) [reply .github/workflows/ci.yml]: All the tests passed now: https://github.com/szabgab/perl-opentracing-implementation-datadog/actions/runs/4626801733

[the remark this answers is not in the document]
Comment (vanHoesel) [reply .github/workflows/ci.yml]: Awesome!!!

**Comment by szabgab** [comment]
I am glad it worked out. BTW There are two test files that are removing these variables and in reality I don't see any tests in this project that would need them.

Anyway, I received a lot of value from the open source work of others. This PR was my way of passing on the good deed to someone else. I am trying now to encourage more people, you too, to find other project(s) and make a small contribution there. See this issue explaining my reasoning further: https://github.com/OSDC-Code-Maven/OSDC-Code-Maven.github.io/issues/5

**Pull request merged**